src/main/java/com/basho/riak/client/core/query/timeseries/FullColumnDescription.java

```diff
@@ -0,0 +1,180 @@
+     *                        <b>Use null if not a key column.</b>
+     * @throws IllegalArgumentException if Column Name or Column Type are null or empty.
+     */
+    public FullColumnDescription(String name,
+                                 ColumnDescription.ColumnType type,
+                                 boolean isNullable,
+                                 Integer partitionKeyOrdinal,
+                                 Integer localKeyOrdinal)
+    {
+        this.baseColumnDescription = new ColumnDescription(name, type);
+        this.isNullable = isNullable;
+        this.partitionKeyOrdinal = partitionKeyOrdinal;
+        this.localKeyOrdinal = localKeyOrdinal;
+    }
+
+    /**
+     * Get the Column Name.
+     * @return the column name String.
+     */
+    public String getName()
+    {
+        return this.baseColumnDescription.getName();
+    }
+
+    /**
+     * Get the Column Type.
+     * @return the ColumnType value.
+     */
+    public ColumnDescription.ColumnType getType()
+    {
+        return this.baseColumnDescription.getType();
+    }
+
+    /**
+     * Whether this column's values are nullable.
+     * @return boolean
+     */
+    public boolean isNullable()
+    {
+        return isNullable;
+    }
+
+    /**
+     * Whether this column is a member of the Partition Key column set.
+     * @return boolean.
+     */
+    public boolean isPartitionKeyMember()
+    {
+        return partitionKeyOrdinal != null;
+    }
+
+    /**
+     * Whether this column is a member of the Local Key column set.
+     * @return boolean.
+     */
+    public boolean isLocalKeyMember()
+    {
+        return localKeyOrdinal != null;
+    }
+
+    /**
+     * Get the ordinal number of where this column appears in the ordered Partition Key column set.
+     * @return Integer if {@link #isPartitionKeyMember()} is <i>true</i>,
+     *         <b>null</b> if {@link #isPartitionKeyMember()} is <i>false</i>.
+     */
+    public Integer getPartitionKeyOrdinal()
+    {
+        return partitionKeyOrdinal;
+    }
+
+    /**
+     * Get the ordinal number of where this column appears in the ordered Local Key column set.
+     * @return Integer if {@link #isLocalKeyMember()} is <i>true</i>,
+     *         <b>null</b> if {@link #isLocalKeyMember()} is <i>false</i>.
+     */
+    public Integer getLocalKeyOrdinal()
+    {
+        return localKeyOrdinal;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o)
+        {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass())
+        {
+            return false;
+        }
+
+        FullColumnDescription that = (FullColumnDescription) o;
+
+        if (isNullable != that.isNullable)
+        {
+            return false;
+        }
+        if (!baseColumnDescription.equals(that.baseColumnDescription))
+        {
+            return false;
+        }
+        if (partitionKeyOrdinal != null ?
```

Comment:
They would be different if you're comparing two rows from different tables and one is a key member, and the other isn't.